File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
       ListNode temp = head;
       int count = 0;
       if(head == null || head.next ==null){
        return null;
       } 
       while(temp!=null){
        temp = temp.next;
        count++;
       }
       if(count ==n){
        return head.next;
       }
       int i =1;
       temp = head;
       while (i<count-n){
        temp =temp.next;
        i++;
       }
       if(temp.next!=null){
        temp.next=temp.next.next;
       }
       return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy_node = ListNode(0,head)
        first=second=dummy_node
        for _ in range(1,n+1):
            first=first.next
        while first.next is not None:
            first=first.next
            second=second.next
        second.next=second.next.next
        return dummy_node.next
            